javascript/javascript2/week1/main.js

```diff
@@ -0,0 +1,135 @@
+const products = getAvailableProducts();
+
+function renderProducts(product) {
+    //your code here
+    
+    for(i=0; i<products.length; i++){
+        const mainList = document.querySelector("ul");
+        const listHeading = document.createElement("h1");
+        const listcomponent1 = document.createElement("li");
+        const listcomponent2 = document.createElement("li");
+        listcomponent1.innerHTML = `price: ${products[i].price}`
+        listcomponent2.innerHTML = `rating: ${products[i].rating}`
+        listHeading.innerHTML = products[i].name
+        mainList.appendChild(listHeading);
+        mainList.appendChild(listcomponent1);
+        mainList.appendChild(listcomponent2);
+        mainList.style.background = "#79a6d2";
+        listHeading.style.color = "purple";
+        listcomponent1.style.color = "white";
+        listcomponent2.style.color = "#ff0000";
+
+    }
+  }
+  renderProducts();
+
+  //Spirited Animal Task
+
+  const spiritAnimals = ["The crying Butterfly", "The fullmoon Wolf", "The cunning Fox", "The roaring Lion", "The flying Hawk", "The aggresive Bear", "The elegant Swan", "The durable Camel", "The peacful Dove", "The lively Dolphin"]
+
+
+const bodyOfPage = document.querySelector("body");
+const nameInput = document.createElement("input");
+nameInput.setAttribute("type", "text");
+nameInput.setAttribute("placeholder", "Write your name here");
+nameInput.setAttribute("value", "");
+nameInput.setAttribute("id", "personName")
+const myButton = document.createElement("button");
+myButton.innerHTML = "Click Me!"
+const resultText = document.createElement("p");
+resultText.style.color = "red";
+resultText.style.fontSize = "16px";
+
+bodyOfPage.appendChild(nameInput);
+bodyOfPage.appendChild(myButton);
+bodyOfPage.appendChild(resultText);
+
+function toggleAnimalName(){
+    const getName = document.getElementById("personName").value;
+    console.log(getName);
+    if(getName.length<=0){
+        resultText.innerHTML = "Please provide your name to see the result";
+        
+        
+    }
+    else{
+        
+        const randomAnimals = spiritAnimals[Math.floor(Math.random()*spiritAnimals.length)];
+        resultText.innerHTML = `${getName}- ${randomAnimals}`
+        
+
+
+    }
+
+}
+
+function buttonClick(){
+    myButton.addEventListener("click", function(){
+        if(document.querySelector("#choice-button").checked){
+            toggleAnimalName();
+        }
+    });
+}
+function hoverEffect(){
```

Comment:
addEventListener will work when you perform the actions, thats when you click, hover, etc. you don't need to enclose addEventListener inside these functions

Reply:
if i remove these functions and just write two addeventlisteners. one with hover and one with click. it will work both at the same time. which i dont want. user will choose one of the given two options and then the function will work accordingly.